test/e2e/app-dir/react-max-headers-length/next.config.js

```diff
@@ -0,0 +1,14 @@
+/**
+ * @type {import('next').NextConfig}
+ */
+const nextConfig = {
+  reactMaxHeadersLength: process.env.TEST_REACT_MAX_HEADERS_LENGTH
+    ? parseInt(process.env.TEST_REACT_MAX_HEADERS_LENGTH)
+    : undefined,
+  experimental: {
+    // Emitting Link headers currently requires the experimental PPR feature.
+    ppr: true,
```

Comment:
technically we do onHeaders for regular dynamic renders but I'm not sure it's worth the test infra to exercise this config value in all these different ways